From f1eec95dd0475591897a781eba15d8d8ce370a6a Mon Sep 17 00:00:00 2001
From: yorke <yorke.yuan@asterfusion.com>
Date: Mon, 3 Jun 2019 13:21:07 +0800
Subject: [PATCH 6/8] [teamd]: Administratively shutdown port channel has
 member ports in deselected state and traffic is not forwarded     From:
 phanindra-tv     https://github.com/Azure/sonic-buildimage/pull/2882

Signed-off-by: yorke <yorke.yuan@asterfusion.com>
---
 teamd/teamd_runner_lacp.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/teamd/teamd_runner_lacp.c b/teamd/teamd_runner_lacp.c
index 9836824..7bf3700 100644
--- a/teamd/teamd_runner_lacp.c
+++ b/teamd/teamd_runner_lacp.c
@@ -1101,12 +1101,17 @@ static int lacpdu_recv(struct lacp_port *lacp_port)
 	struct lacpdu lacpdu;
 	struct sockaddr_ll ll_from;
 	int err;
+	bool admin_state;
 
 	err = teamd_recvfrom(lacp_port->sock, &lacpdu, sizeof(lacpdu), 0,
 			     (struct sockaddr *) &ll_from, sizeof(ll_from));
 	if (err <= 0)
 		return err;
 
+	admin_state = team_get_ifinfo_admin_state(lacp_port->ctx->ifinfo);
+	if (!admin_state)
+		return 0;
+
 	if (!teamd_port_present(lacp_port->ctx, lacp_port->tdport))
 		return 0;
 
-- 
2.7.4

